src/Result.php

```diff
@@ -6,20 +6,34 @@
 
 final class Result
 {
+    /**
+     * @var ErrorMessage[]
+     */
     private array $errors = [];
 
     public function isValid(): bool
     {
         return $this->errors === [];
     }
 
-    public function addError(string $message): void
+    public function addError(ErrorMessage $error): void
     {
-        $this->errors[] = $message;
+        $this->errors[] = $error;
     }
 
-    public function getErrors(): array
+    /**
+     * @return ErrorMessage[]
```

Comment:
```suggestion
     * @return string[]
```

Reply:
No, this method return array with instances of ErrorMessage

Reply:
The same

Reply:
It's function getRawErrors()